### PR TITLE
CMake fixes

### PR DESCRIPTION
**Pull request by Ninja3047**
* move doctest to cxx-common: lifting-bits/cxx-common#964 and remove submodule from repo
* update git_watcher.cmake to fix some bugs, most notably changing `CMAKE_BINARY_DIR` to `CMAKE_CURRENT_BINARY_DIR`
* remove llvm.cmake and janky find_llvm, use upstream llvm recommended cmake incantation
* update FindZ3.cmake to fix some bugs, most notably changing `CMAKE_SOURCE_DIR` a temporary path

### CODE REVIEW

**Comment by Ninja3047** [comment]
won't build until  lifting-bits/cxx-common#964 gets merged and tagged so do not merge until then

**Review by frabert — APPROVED** [review]

**Pull request merged**
